lib/vrf.c

```diff
@@ -966,3 +966,11 @@ int vrf_sockunion_socket(const union sockunion *su, vrf_id_t vrf_id,
 	}
 	return ret;
 }
+
+void vrf_list_walk(void (*exec_for_each_vrf)(struct vrf *))
```

Comment:
Is this a list?  Why are we calling it this?  Shouldn't this just be vrf_walk?

Reply:
Additionally why are we using RB_FOREACH sometimes and sometimes using vrf_walk?  When should I use one over the other? Can you help me understand?